src/v/transform/rpc/deps.cc

```diff
@@ -30,6 +30,7 @@
 #include <seastar/core/future.hh>
```

Comment:
> transform/rpc: prevent cross shard memory writes
>
> I'm not fully sure if this is safe with respect to local thread caches, so
instead use the proper seastar mechanism to pass data between shards.

what's the concern, the change, and the rationale for it addressing the concern?

Reply:
Updated the cover letter and commit message to expand on the motivation.